escheduler-server/src/main/java/cn/escheduler/server/worker/task/http/HttpTask.java

```diff
@@ -0,0 +1,310 @@
+                }
+                break;
+            case BODY_NOT_CONTAINS:
+                if (StringUtils.isEmpty(body) || body.contains(httpParameters.getCondition())) {
+                    appendMessage(httpParameters.getUrl() + " contains "
+                            + XmlEscapers.xmlContentEscaper().escape(httpParameters.getCondition()));
+                    exitStatusCode = -1;
+                }
+                break;
+            case STATUS_CODE_CUSTOM:
+                if (!statusCode.equals(httpParameters.getCondition())) {
+                    appendMessage(httpParameters.getUrl() + " statuscode: " + statusCode + ", Must be: " + httpParameters.getCondition());
+                    exitStatusCode = -1;
+                }
+                break;
+            default:
+                if (!"200".equals(statusCode)) {
+                    appendMessage(httpParameters.getUrl() + " statuscode: " + statusCode + ", Must be: 200");
+                    exitStatusCode = -1;
+                }
+                break;
+        }
+        return exitStatusCode;
+    }
+
+    public String getOutput() {
+        return output;
+    }
+
+    protected void appendMessage(String message) {
+        if (output == null) {
+            output = "";
+        }
+        if (message != null && !message.trim().isEmpty()) {
+            output += message;
+        }
+    }
+
+    protected void addRequestParams(RequestBuilder builder,Map<String, Property> paramsMap) {
+        Iterator<Map.Entry<String, Property>> iter = paramsMap.entrySet().iterator();
+        while (iter.hasNext()){
+            Map.Entry<String, Property> en = iter.next();
+            Property property = en.getValue();
+            if (property.getValue() != null && property.getValue().length() > 0){
+                if (property.getDirect().equals(Direct.IN)){
+                    builder.addParameter(property.getProp(), property.getValue());
+                }else{
+                    if(CONTENT_TYPE.equals(property.getProp()) && property.getValue().contains(APPLICATION_JSON)){
+                        StringEntity postingString = null;
+                        try {
+                            postingString = new StringEntity("{}");
+                        } catch (UnsupportedEncodingException e) {
+                            e.printStackTrace();
+                        }
+                        builder.setEntity(postingString);
+                    }
+                }
+            }
+        }
+    }
+
+    protected void setHeaders(HttpUriRequest request,Map<String, Property> paramsMap) {
+        Iterator<Map.Entry<String, Property>> iter = paramsMap.entrySet().iterator();
+        while (iter.hasNext()){
+            Map.Entry<String, Property> en = iter.next();
+            Property property = en.getValue();
+            if (property.getValue() != null && property.getValue().length() > 0){
+                if (property.getDirect().equals(Direct.OUT)){
```

Comment:
Don't know why we treat OUT parameter as Http Header, while IN parameter as Query Parameter?    Seems you reuse old SQL Procedure logic for IN/OUT parameter. but For Http, we need to change the Parameter Type to "Header", "Query String", etc.   Otherwise, later maintainer may be very confused about this IN/OUT parameter.